packages/react-portal/Spec.md

```diff
@@ -0,0 +1,201 @@
+```
+
+`Portal` should have a fallback of document.body if `PortalProvider` is not used. This fallback should still be able to access theme and fluent context if available
+
+```
+<App> // using FluentProvider of ThemeProvider but not PortalProvider
+  <Portal /> // attached to document.body
+  <Portal /> // attached to document.body
+</App>
+```
+
+`Portal` should be used as a component at any part of the React tree
+
+```tsx
+<ContextProvider>
+  <Portal>
+    <ContextConsumer /> // should be able to access context
+  </Portal>
+</ContextProvider>
+```
+
+`Portal` should be able to access theme values as css variables
+
+```tsx
+const useStyles = makeStyles({
+    portalContent: theme => {...}
+})
+
+
+const styles = useStyles();
+
+<ThemeProvider>
+    <Portal className={styles.portalContent>
+      Can use all theme CSS variables from the parent ThemeProvider
+    </Portal>
+</ThemeProvider>
+```
+
+## Variants
+
+- Mounting the portal on a custom node
+- Disable event bubbling
+
+## API
+
+### Portal
+> TODO clear whether `onMount` or `onUnmount` need to be called when mount node changes like in v8
+>
+> TODO clear whether `insertionOrder` should be supported in convergence
+>
+> TODO clear whether `disableEventBubbling` should be supported in convergence
+
+| Name                 | Description                                                   | Required | Type              | Default value                    |
+| -------------------- | ------------------------------------------------------------- | -------- | ----------------- | -------------------------------- |
+| mountNode            | Where the portal is mounted to the DOM                        | No       | HTMLElement       | ProviderContext or document.body |
+| onMount              | Called when the portal is mounted                             | No       | Function          |                                  |
+| onUnmount            | Called when the portal is unmounted                           | No       | Function          |                                  |
+| disableEventBubbling | Disables event bubbling to the React tree                     | No       | Boolean           |                                  |
+| insertionOrder       | Position of the portal content in the mountNode               | No       | 'first' \| 'last' | 'last'                           |
+
+### PortalContext
+
+Context that store a mount node passed to portals below the provider in the React tree, can be easily accessed with `React.useContext`;
+
+| Name  | Description                       | Required | Type        |
+| ----- | --------------------------------- | -------- | ----------- |
+| value | The DOM element to insert portals | Yes      | HTMLElement |
+
+## Structure
+
+Usage with provider
+
+```
+<FluentProvider
+  <ThemeProvider>
+    <PortalProvider>
+      <Portal id="portal-1" />
+      <Portal id="portal-2" />
+    </PortalProvider>
+  </ThemeProvider>
+</FluentProvider
```

Comment:
What's the difference between `FluentProvider` and `ThemeProvider` here?

Reply:
It's supposed to show that both `FluentProvider` and `ThemeProvider` can apply theme... but you're right and it's confusing... will remove

```suggestion
<FluentProvider
  <PortalProvider>
    <Portal id="portal-1" />
    <Portal id="portal-2" />
  </PortalProvider>
</FluentProvider
```